src/connection.rs

```diff
@@ -306,6 +306,51 @@ impl<S, C> DeviceConnection<S, C> {
         Ok(true)
     }
 
+    async fn remove_interfaces<'a, I>(&mut self, interfaces_name: I) -> Result<Vec<String>, Error>
+    where
+        C: Register,
+        S: PropertyStore,
+        I: IntoIterator<Item = &'a str> + Send,
+    {
+        let mut interfaces = self.interfaces.write().await;
+
+        let to_remove: HashMap<&str, &Interface> = interfaces_name
+            .into_iter()
+            .filter_map(|iface_name| {
+                let interface = interfaces.get(iface_name).map(|i| (i.interface_name(), i));
+
+                if interface.is_none() {
+                    debug!("{iface_name} not found, skipping");
+                }
+
+                interface
+            })
+            .collect();
+
+        if to_remove.is_empty() {
+            return Ok(Vec::new());
+        }
+
+        self.connection
+            .remove_interfaces(&interfaces, &to_remove)
+            .await?;
+
+        for (_, iface) in to_remove.iter() {
+            // We cannot error here since we already unsubscribed to the interface
```

Comment:
This comment is referring to the inner if, also I wrote it in an awful English 🙇🏻. If you can reword  this and the above one

```suggestion
            // We cannot error here since we have already unsubscribed from the interface
```